tests/validation/tests/v3_api/test_upgrade.py

```diff
@@ -457,11 +460,21 @@ def create_validate_wokloads_with_secret():
 @pytest.fixture(scope='module', autouse="True")
 def create_project_client(request):
     client = get_user_client()
+    admin_client = get_admin_client()
     clusters = client.list_cluster(name=cluster_name).data
     assert len(clusters) == 1
     cluster = clusters[0]
     create_kubeconfig(cluster)
     namespace["cluster"] = cluster
+    if len(admin_client.list_catalog(name="test-catalog")) == 0:
+        catalog = admin_client.create_catalog(
+            name="test-catalog",
+            baseType="catalog",
+            branch=catalogBranch,
+            kind="helm",
+            url=catalogUrl)
+        time.sleep(5)
```

Comment:
@sowmyav27 Can we wait for catalog state to get to active , instead of a sleep for 5 sec? Keep the max wait for 15 sec just to be sure.

Reply:
Added function wait_for_catalog_active in common.py